src/stores/Create.ts

```diff
@@ -297,21 +317,30 @@ export const getInitialState = (
         prefillCollections: (value) => {
             set(
                 produce((state) => {
-                    state.collections = value;
-                    state.resourceConfig = {};
+                    const collections: string[] = [];
+                    const configs = {};
 
                     value.forEach((collection) => {
-                        state.resourceConfig[collection] =
-                            getDefaultJsonFormsData();
+                        collection.writes_to.forEach((writes_to) => {
+                            collections.push(writes_to);
+                            configs[writes_to] = getDefaultJsonFormsData();
+                        });
                     });
+
+                    state.collections = collections;
```

Comment:
If `state.collections` can be set using with a string array, then the types of this property and the input argument of `state.setCollections` can be safely narrowed at this time.

Reply:
updated